test/build/50_ch-image.bats

```diff
@@ -677,6 +677,96 @@ EOF
     ch-image delete tmpimg:tagged
 }
 
+@test 'ch-image build: multistage with --target' {
+    df=$(cat <<'EOF'
+FROM alpine:3.17 as aioli
+RUN touch stage0
+FROM alpine:3.16 as marinara
+RUN touch stage1
+FROM scratch
+COPY --from=0 /stage0 /
+EOF
+)
+    # No multistage or target.
+    img="${CH_IMAGE_STORAGE}/img/multisauce"
+    run ch-image build -t multisauce -f - . <<EOF
+${df}
+EOF
+    echo "$output"
+    [[ $status -eq 0 ]]
+    [[ $output == *'alpine:3.17 AS aioli'* ]]
+    [[ $output == *'FROM alpine:3.16 AS marinara'* ]]
+    [[ $output == *'FROM scratch'* ]]
+    [[ $output == *'grown in 6 instructions: multisauce'* ]]
+    test -f "${img}/stage0"
+    test ! -f "${img}/stage1"
+
+    # Digit as argument.
+    run ch-image build --target=1 -t multisauce -f - . <<EOF
+${df}
+EOF
+    echo "$output"
+    [[ $status -eq 0 ]]
+    [[ $output == *'FROM alpine:3.17 AS aioli'* ]]
+    [[ $output == *'FROM alpine:3.16 AS marinara'* ]]
+    [[ $output == *'grown in 4 instructions: multisauce'* ]]
+    test -f "${img}/stage1"
+    test ! -f "${img}/stage0"
+
+    # Alias as argument.
+    run ch-image build --target=aioli -t multisauce -f - . <<EOF
+${df}
+EOF
+    echo "$output"
+    [[ $status -eq 0 ]]
+    [[ $output == *'FROM alpine:3.17 AS aioli'* ]]
+    [[ $output == *'grown in 2 instructions: multisauce'* ]]
+    test -f "${img}/stage0"
+    test ! -f "${img}/stage1"
+
+    # Base as argument.
+    run ch-image build --target=alpine:3.16 -t multisauce -f - . <<EOF
+${df}
+EOF
+    echo "$output"
+    [[ $status -eq 0 ]]
+    [[ $output == *'FROM alpine:3.17 AS aioli'* ]]
+    [[ $output == *'FROM alpine:3.16 AS marinara'* ]]
+    [[ $output == *'grown in 4 instructions: multisauce'* ]]
+    test -f "${img}/stage1"
+    test ! -f "${img}/stage0"
+
+    # Error: target is last stage.
```

Comment:
I don't think this should be an error. Accepting the last stage as a target will make it easier for anyone generating Charliecloud command lines b/c they won't have to deal with it as a special case.

Similarly, I think we should accept `--target` for single-stage images.